### PR TITLE
Migrate to AndroidX

### PR DESCRIPTION
**Pull request by octohub**
Handles: https://github.com/Raizlabs/android-template/issues/24

### CODE REVIEW

File: {{ cookiecutter.repo_name }}/gradle/dependencies.gradle [hunk 1]
```diff
@@ -29,41 +29,41 @@ versions.mockito        = '2.22.0'
...
-                        'recyclerView': "com.android.support:recyclerview-v7:${versions.supportLibrary}",
-                        'constraintLayout':'com.android.support.constraint:constraint-layout:2.0.0-alpha2'
+                        'annotations': "androidx.annotation:annotation:${versions.supportLibrary}",
+                        'appcompat': "androidx.appcompat:appcompat:${versions.supportLibrary}",
+                        'cardview': "androidx.cardview:cardview:${versions.supportLibrary}",
+                        'compat': "androidx.core:core:${versions.supportLibrary}",
+                        'coreUi': "androidx.legacy:legacy-support-core-ui:${versions.supportLibrary}",
+                        'coreUtils': "androidx.legacy:legacy-support-core-utils:${versions.supportLibrary}",
+                        'design': "com.google.android.material:material:${versions.supportLibrary}",
+                        'fragment': "androidx.fragment:fragment:1.0.0:${versions.supportLibrary}",
```
Comment (arjunvekariyagithub) [inline {{ cookiecutter.repo_name }}/gradle/dependencies.gradle]: #rq need to remove `1.0.0`

**Review by arjunvekariyagithub — CHANGES_REQUESTED** [review]
Minor change, otherwise looks good

File: {{ cookiecutter.repo_name }}/gradle/dependencies.gradle [hunk 2]
```diff
@@ -14,12 +14,12 @@ ext.buildConfig         = buildConfig
...
+versions.supportLibrary = '1.0.0'
 versions.arch           = '1.1.1'
-versions.archRoom       = '1.1.1'
-versions.atsl           = '1.0.2'
-versions.espresso       = '3.0.2'
-versions.dagger         = '2.17'
+versions.archRoom       = '2.0.0'
+versions.atsl           = '1.1.0-alpha3'
+versions.espresso       = '3.1.0-alpha3'
+versions.dagger         = '2.16'
```
Comment (arjunvekariyagithub) [inline {{ cookiecutter.repo_name }}/gradle/dependencies.gradle:22]: any reason to downgrade `dagger`?

Comment (octohub) [reply {{ cookiecutter.repo_name }}/gradle/dependencies.gradle:22]: Yes, there is currently a known issue with AndroidX and Dagger 2.17, documented here:
https://github.com/google/dagger/issues/1245
https://issuetracker.google.com/issues/115738511

We will update Dagger once the issue is resolved.

Comment (arjunvekariyagithub) [reply {{ cookiecutter.repo_name }}/gradle/dependencies.gradle:22]: I see. Thanks.

**Review by arjunvekariyagithub — APPROVED** [review]

**Pull request merged**